wordpress-mobile/WordPress-iOS
# Full Offline Publishing Experience

Initial step towards implementing #12227

This PR implements the following changes:

- Changes the messages seen in the notices when you save posts while offline (a different message for saved / published / scheduled posts).
- Changes the "Retry" button in the post list to a "Cancel" button, and the logic for it.
- Changes the message and the color of the message shown in the post list for posts with unsaved changes and for posts that were saved while offline.
- Fixes a bug in the post list that would hide the Retry / Cancel button.  This bug was introduced by some previous changes in `develop` but was particularly annoying and evident while testing this PR.
- Modifies notice views so that they can show 2 line titles. (fyi @frosty)
- Modifies post cards so that they can show 2 line messages.

## Demo:

![offlinePublishing](https://user-images.githubusercontent.com/1836005/61653983-e0026700-ac91-11e9-863d-ab2106cfade0.gif)

## Scope:

This PR doesn't aim to implement all of the changes we need for #11425.  This PR aims to implement the basic messaging changes, some UI fixes, and replacing the "Retry" logic with a "Cancel" logic.

## Testing:

### Test 1: Notices

Try editing or creating a post while offline, and save the changes.
The notice shown should read:

"Post will be saved / scheduled / published the next time your device is online"

### Test 2: Cancelling

1. Try editing or creating a post while offline, and save the changes.
2. Try pressing "Cancel" in the notice that comes up.
3. Alternatively try pressing "Cancel" in the post list.

Make sure the post upload is cancelled and that the post is listed as having local changes.

### Test 3: Post list text color

1. Try editing or creating a post while offline, and save the changes.
2. Make sure the text color is yellow (warning) in the post list.
3. Make sure the button color for "Cancel" is yellow (warning) in the post list.
4. Tap "Cancel" in the post list.

Make sure the text is now red (error).  Also make sure the text reads "Post has local unsaved changes"

## Update release notes:
